server/datastore/mysql/vpp.go

```diff
@@ -373,7 +425,10 @@ ON DUPLICATE KEY UPDATE
 		}
 	}
 
-	return ctxerr.Wrap(ctx, err, "writing vpp app team mapping to db")
+	id, _ := res.LastInsertId()
```

Comment:
Does this work on the ON DUPLICATE KEY UPDATE path?

Reply:
Good catch; there's a pattern to deal with this that I've added here.